components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/artifact/SAMLSSOSoapMessageService.java

```diff
@@ -0,0 +1,160 @@
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLContextBuilder;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.BasicResponseHandler;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.opensaml.common.SAMLObject;
+import org.opensaml.ws.soap.common.SOAPObjectBuilder;
+import org.opensaml.ws.soap.soap11.Body;
+import org.opensaml.ws.soap.soap11.Envelope;
+import org.opensaml.xml.Configuration;
+import org.opensaml.xml.XMLObjectBuilderFactory;
+import org.wso2.carbon.identity.sso.agent.exception.ArtifactResolutionException;
+import org.wso2.carbon.identity.sso.agent.util.SSOAgentConstants;
+
+import java.io.IOException;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * This class is used for handling SAML SOAP Binding
+ */
+public class SAMLSSOSoapMessageService {
+
+    private static final String CONTENT_TYPE = "text/xml; charset=utf-8";
+    private static final String MIME_TYPE = "text/xml";
+    private static Log log = LogFactory.getLog(SAMLSSOSoapMessageService.class);
+
+    /**
+     * Build a SOAP Message.
+     *
+     * @param samlMessage SAMLObject.
+     * @return Envelope soap envelope
+     */
+    public Envelope buildSOAPMessage(SAMLObject samlMessage) {
+
+        XMLObjectBuilderFactory builderFactory = Configuration.getBuilderFactory();
+
+        SOAPObjectBuilder<Envelope> envBuilder = (SOAPObjectBuilder<Envelope>) builderFactory.getBuilder(
+                Envelope.DEFAULT_ELEMENT_NAME);
+        Envelope envelope = envBuilder.buildObject();
+
+        SOAPObjectBuilder<Body> bodyBuilder = (SOAPObjectBuilder<Body>) builderFactory.getBuilder(
+                Body.DEFAULT_ELEMENT_NAME);
+        Body body = bodyBuilder.buildObject();
+        body.getUnknownXMLObjects().add(samlMessage);
+        envelope.setBody(body);
+        return envelope;
+    }
+
+    /**
+     * Send SOAP message
+     *
+     * @param message message that needs to be send
+     * @param url     url that the artifact resolve request should be sent
+     * @return response of invoking artifact resolve endpoint
+     * @throws ArtifactResolutionException
+     */
+    public String sendSOAP(String message, String url) throws ArtifactResolutionException {
+
+        StringBuilder soapResponse = new StringBuilder();
```

Comment:
Validate whether message and url is not null. would be better to print those in a debug log.